sherpa/sbom.go

```diff
@@ -0,0 +1,154 @@
+package sherpa
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/CycloneDX/cyclonedx-go"
+	"github.com/buildpacks/libcnb"
+	"github.com/paketo-buildpacks/libpak/bard"
+	"github.com/paketo-buildpacks/libpak/effect"
+)
+
+//go:generate mockery -name BOMScanner -case=underscore
+
+type BOMScanner interface {
+	Scan(location libcnb.BOMLocation, scanDir string, formats ...libcnb.BOMFormat)
+}
+
+type SyftCLIBOMScanner struct {
+	Layer    libcnb.Layer
+	Executor effect.Executor
+	Logger   bard.Logger
+}
+
+// Scan will use syft CLI to scan the scanDir and write it's output to the location in the given formats
+func (b SyftCLIBOMScanner) Scan(location libcnb.BOMLocation, scanDir string, formats ...libcnb.BOMFormat) error {
+	// syft doesn't presently support outputting multiple formats at once
+	// to workaround this we are running syft multiple times
+	// when syft supports multiple output formats or conversion between formats, this method should change
+	for _, format := range formats {
+		if err := b.runSyft(location, scanDir, format); err != nil {
+			return fmt.Errorf("unable to run syft\n%w", err)
+		}
+
+		if format == libcnb.CycloneDXJSON {
+			// syft doesn't presently support cyclonedx JSON output and we need to convert
+			// until https://github.com/anchore/syft/issues/631 is addressed
+			if err := b.ConvertCycloneDXXMLtoJSON(location, false); err != nil {
+				return fmt.Errorf("unable convert XML to JSON\n%w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// ConvertCycloneDXXMLtoJSON reads input CycloneDX XML, converts to JSON and overwrites the XML optionally keeping a backup copy of the xml
+func (b SyftCLIBOMScanner) ConvertCycloneDXXMLtoJSON(location libcnb.BOMLocation, backup bool) error {
+	inputPath := b.Layer.BOMPath(location, libcnb.CycloneDXJSON)
+
+	if backup {
+		if err := b.backupXMLFile(inputPath); err != nil {
+			return fmt.Errorf("unable to backup file\n%w", err)
+		}
+	}
+
+	bom, err := b.readXMLBOM(inputPath)
+	if err != nil {
+		return fmt.Errorf("unable to backup file\n%w", err)
+	}
+
+	if err := b.writeJSONBOM(inputPath, bom); err != nil {
+		return fmt.Errorf("unable to backup file\n%w", err)
```

Comment:
unable to write converted JSON BOM file?

Reply:
```suggestion
		return fmt.Errorf("unable to write converted JSON BOM file\n%w", err)
```